src/libs/state.ts

```diff
@@ -25,6 +25,13 @@ export type TerraUserProfile = {
   interestInTerra: string | undefined;
 };
 
+export type TerraUserStatus = 'unregistered' | 'registeredWithoutTos' | 'registered' | 'disabled' | 'uninitialized';
```

Comment:
"Status" is fairly generic. Could we call this "RegistrationStatus" or something like that? Also, (not necessarily in this PR) it would be nice to document what these different statuses mean.

```suggestion
export type TerraUserRegistrationStatus = 'unregistered' | 'registeredWithoutTos' | 'registered' | 'disabled' | 'uninitialized';
```

```suggestion
export type TerraUserRegistrationStatus =
  // User is logged in through B2C but has not registered in Terra.
  | 'unregistered'
  // User has registered in Terra but has not accepted the terms of service.
  | 'registeredWithoutTos'
  // User has registered in Terra and accepted the terms of service.
  | 'registered'
  // User's account has been disabled.
  | 'disabled'
  // Registration status has not yet been determined.
  | 'uninitialized';
```

Reply:
I guess I had a hard time naming this since `'disabled'` I would not exactly consider a registration status. But since the logic they are used in is the same areas, I think it should not be confusing.

Reply:
>  since 'disabled' I would not exactly consider a registration status

🤔 Yeah, that makes sense. I'm on board with leaving it as "TerraUserStatus" then.

Reply:
Maybe it could be changed to `TerraUserPermissionStatus` or `TerraUserAuthorizationStatus` just because it is being used to determine where the user is allowed to go on Terra?

Reply:
I was thinking "RegistrationStatus" because it's the type of the `registrationStatus` field in the auth store.

I'm not too keen on "permission" or "authorization" because most permissions in Terra are at the workspace level instead of at the user level / global.

Reply:
Ill put it as `TerraUserRegistrationStatus` for now to match the authStore. 

I do think that a renaming would be good at some point for both the type and the authStore field, but I think that will be better to change as we do our ToS work later in the quarter.